app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -364,6 +376,37 @@ private void receiveSharedItems() {
                     .getQuantityString(R.plurals.upload_count_title, uploadableFiles.size(), uploadableFiles.size()));
 
             fragments = new ArrayList<>();
+            /* Suggest users to turn battery optimisation off when uploading more than a few files.
+               That's because we have noticed that many-files uploads have
+               a much higher probability of failing than uploads with less files.
+             */
+            if (uploadableFiles.size() > 3
+                && !defaultKvStore.getBoolean("hasAlreadyLaunchedBigMultiupload")) {
```

Comment:
I guess we should show this dialog only for relevant Android versions. The intent itself is only available since API 23 (6+). Kindly check and make sure we only invoke this for the relevant Android versions.